src/arm7tdmi.rs

```diff
@@ -277,4 +300,18 @@ mod tests {
             assert_eq!(cpu.registers[rx as usize], rx.rotate_right(is * 2));
         }
     }
+
+    #[test]
+    fn check_teq() {
+        let op_code: u32 = 0b1110_0001_0010_1001_0011_0000_0000_0000;
+        let mut cpu = Arm7tdmi::new(vec![]);
+
+        let (_, instruction) = cpu.decode(op_code);
+        assert_eq!(instruction, ArmModeInstruction::DataProcessing1);
+
+        cpu.execute(op_code, instruction);
+        let rn = 0b0011;
+        let rd_xor_op2 = 0b1001;
+        assert_eq!(cpu.registers[rn], rd_xor_op2);
```

Comment:
For the above reason this test should be different.

Reply:
The right name is `rd` I'm updating it